### PR TITLE
Add gtest for writing unit tests

### PR DESCRIPTION
**Pull request by LukasWoodtli**
gtest will be used for writing unit test for sitl_gazebo.

### CODE REVIEW

**Comment by TSC21** [comment]
Is `libgtest-dev` intalled?

File: docker/px4-dev/Dockerfile_simulation [hunk 1]
```diff
@@ -14,14 +14,19 @@ RUN wget --quiet http://packages.osrfoundation.org/gazebo.key -O - | apt-key add
...
 		libimage-exiftool-perl \
 		libopencv-dev \
 		libxml2-utils \
 		pkg-config \
 		protobuf-compiler \
 	&& apt-get -y autoremove \
 	&& apt-get clean autoclean \
-	&& rm -rf /var/lib/apt/lists/{apt,dpkg,cache,log} /tmp/* /var/tmp/*
+	&& rm -rf /var/lib/apt/lists/{apt,dpkg,cache,log} /tmp/* /var/tmp/* \
+	# gtest
```
Comment (TSC21) [inline docker/px4-dev/Dockerfile_simulation]: Can you put this on a separate `RUN`?

Comment (LukasWoodtli) [reply docker/px4-dev/Dockerfile_simulation]: Yes of course. But it will create a new layer in Docker...

Comment (TSC21) [reply docker/px4-dev/Dockerfile_simulation]: not a problem

**Comment by LukasWoodtli** [comment]
Can the commits be squashed to have a cleaner git history or should a create a new PR?

**Comment by dagar** [comment]
We can squash of merge.

**Comment by dagar** [comment]
@TSC21 yes adding to base should be fine.

**Comment by TSC21** [comment]
@dagar can we merge then?

**Comment by dagar** [comment]
Wait, this might be my own ignorance, but is this really how gtest is normally used?

Can't the actual project find it from libgtest-dev ,then build and link if needed?

**Comment by LukasWoodtli** [comment]
This seems to be the approach to install gtest as a system library on Ubuntu.
There are other ways like explained here: https://github.com/google/googletest/blob/master/googletest/README.md
The system library approach works well as long as a project doesn't use incompatible compiler/linker flags and doesn't require a lot of CMake stuff (just FindGTest).

**Comment by dagar** [comment]
Can you delete all the other build artifacts?

**Comment by TSC21** [comment]
@dagar can we bring this in?

File: docker/px4-dev/Dockerfile_base [hunk 2]
```diff
@@ -47,6 +48,13 @@ RUN apt-get update && apt-get -y --quiet --no-install-recommends install \
 	&& apt-get clean autoclean \
 	&& rm -rf /var/lib/apt/lists/{apt,dpkg,cache,log} /tmp/* /var/tmp/*
 
+# gtest
+RUN cd /usr/src/gtest \
+	&& mkdir build && cd build \
+	&& cmake CMakeLists.txt .. && make \
```
Comment (dagar) [inline docker/px4-dev/Dockerfile_base]: `cmake CMakeLists.txt ..` -> `cmake ..`

Comment (TSC21) [reply docker/px4-dev/Dockerfile_base]: https://www.eriksmistad.no/getting-started-with-google-test-on-ubuntu/ yeah actually we don't need the build dir here. @LukasWoodtli

Comment (LukasWoodtli) [reply docker/px4-dev/Dockerfile_base]: I know. But it’s easier to remove unneeded build artefacts. It’s just a matter of removing the build directory after moving the needed build files...

Comment (TSC21) [reply docker/px4-dev/Dockerfile_base]: Ok so better to apply the suggestion from @dagar

Comment (dagar) [reply docker/px4-dev/Dockerfile_base]: All I'm suggesting is dropping the unnecessary `CMakeLists.txt`. When running cmake you just need to point it at the source directory.

Comment (LukasWoodtli) [reply docker/px4-dev/Dockerfile_base]: @dagar Sorry I somehow missed your review...

**Pull request merged**